sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionClientBuilder.java

```diff
@@ -5,21 +5,22 @@
 
 
 import com.azure.core.annotation.ServiceClientBuilder;
+import com.azure.core.cryptography.KeyEncryptionKeyResolver;
 import com.azure.cosmos.CosmosAsyncClient;
 import com.azure.cosmos.CosmosClient;
-import com.azure.cosmos.encryption.keyprovider.EncryptionKeyWrapProvider;
 
 /**
  * Helper class to build {@link CosmosEncryptionAsyncClient} and {@link CosmosEncryptionClient}
  * instances as logical representation of the Azure Cosmos database service.
  * <p>
- * When building client, cosmosAsyncClient()/cosmosClient() and encryptionKeyWrapProvider() are mandatory APIs, without these the initialization will fail.
+ * When building client, cosmosAsyncClient()/cosmosClient(), keyEncryptionKeyResolver() and keyEncryptionKeyResolverName() are mandatory APIs, without these the initialization will fail.
  * <pre>
  *     Building Cosmos Encryption Async Client APIs
  * {@code
  * CosmosEncryptionAsyncClient cosmosEncryptionAsyncClient = new CosmosEncryptionClientBuilder()
  *         .cosmosAsyncClient(cosmosAsyncClient)
- *         .encryptionKeyWrapProvider(encryptionKeyWrapProvider)
+ *         .keyEncryptionKeyResolver(keyEncryptionKeyResolver)
```

Comment:
My 2 cents - i would make this one method (.keyEncryptionResolver(reolver, resolverName)

The resolver is a required parameter anyway

But if that violates the current pattern int he builders feel free to ignore my comment. I just find it super unintuitive to identify which combinations of parameters are required/possible etc. in the builder pattern

Reply:
Yea we have builder method where we have error handing for this

Reply:
hmm, should the KeyResolver indicate the resolver name?